C_CPP/Pointer and its types/Pointers and Functions/pointers_to_functions.md

```diff
@@ -0,0 +1,96 @@
+# Pointers and Functions
+
+In C/C++ programming we can pass and return pointers just like normal variables.
+
+Here are some ways in which pointers and functions can be used in C/C++.
+
+## Pointers as function arguments
+<hr>
+
+We can use pointers in functions such that the data passed to the function is reflected back in the main program.
+
+#### Following is a simple program to swap two numbers which implements this:
+
+```c++
+#include<iostream>
+using namespace std;
+void swap(int *x,int *y){
+ int temp;
+ temp = *x;
+ *x = *y;
+ *y = temp;
+}
+int main(){
+    int a = 10,b = 20;
+    cout<<"The numbers before swapping are:"<<a<<" and "<<b<<endl;
+    swap(&a,&b);
+    cout<<"The numbers after swapping are:"<<a<<" and "<<b<<endl;
+    return 0;
+}
+ ```
+### Output
+```
+The numbers before swapping are:10 and 20
+The numbers after swapping are:20 and 10
+```
+<!-- ![Diagram1](https://user-images.githubusercontent.com/77008381/134775453-0d4c7f86-98bb-46ff-aeb1-8a6afc16d0e3.jpg) -->
+<p align = "center"><img src = "https://user-images.githubusercontent.com/77008381/134775453-0d4c7f86-98bb-46ff-aeb1-8a6afc16d0e3.jpg"></p>
+## Functions returning Pointer
```

Comment:
Add a new line before this :
Pro-tip : Whenever you add a html element always add a new line before using md syntax because otherwise it is not rendered...
![image](https://user-images.githubusercontent.com/63473496/134816517-c437b3f1-f643-492c-bf45-1e3d91fa017c.png)

(PS: I like the fact that you've embedded images but used html tags to center them... Good 👍🏻 )

Reply:
```suggestion

## Functions returning Pointer
```